### ISSUE TITLE
GPU mode yield inaccurate result

### ISSUE THREAD

**Issue by CuberMessenger** [Forceflow/cuda_voxelizer#61]
I just voxelized a simple cube, defined as (obj file):
v 150.100006 400.100006 550.099976
v 550.099976 400.100006 550.099976
v 150.100006 800.099976 550.099976
v 550.099976 800.099976 550.099976
v 150.100006 800.099976 150.100006
v 550.099976 800.099976 150.100006
v 150.100006 400.100006 150.100006
v 550.099976 400.100006 150.100006
# 8 vertices, 0 vertices normals

f 1 2 4
f 1 4 3
f 3 4 6
f 3 6 5
f 5 6 8
f 5 8 7
f 7 8 2
f 7 2 1
f 2 8 6
f 2 6 4
f 7 1 3
f 7 3 5
# 12 faces, 0 coords texture

The CPU mode generate fine result, but the GPU  mode yield something inaccurate:
<img width="617" alt="image" src="https://user-images.githubusercontent.com/27040471/144178398-4cd24253-5489-46a0-a834-65c7b75406fd.png">
![31](https://user-images.githubusercontent.com/27040471/144178468-5f6149fd-874d-49ea-aba1-25e6832b424d.png)
![32](https://user-images.githubusercontent.com/27040471/144178481-20119cc5-71ac-4a42-b23a-a488325a6696.png)
![33](https://user-images.githubusercontent.com/27040471/144178444-abdb9ec1-bf7e-4d6c-8181-9f212de15673.png)
The first image is the voxelization result visualized by marching cubes algorithm and meshLab, some glitch appears along the diagonal line. The latter 3 images are 3 continuous 2-D plane of the voxelization result, the middle one shows that the glitches comes from the voxelizer.

I barely know about cuda programming and the voxelization algorithm, just point out this issue, maybe someone can fix it.
Still very nice voxelizer 😃 !

**Comment by Forceflow** [comment]
Yeah, I think this is an artifact of the bounding box of the model perfectly aligning with the model itself. Probably a numeric instability. Will look into it.
